File: Pubsub_ws/Example_py_ws/Example_py_ws/sub.py
import rclpy
from rclpy.node import Node

#----------DECLARATIONS----------
from std_msgs.msg import UInt8MultiArray

class Alpha_Subscriber(Node):
    #----------NODE DECLARATION----------
    def __init__(self):
        super().__init__('Alpha_Subscriber')
        self.subscriber_ = self.create_subscription(UInt8MultiArray, 'Topic_2', self.listener_callback, 10)
        self.subscriber_

    #----------SUBSCRIBER----------
    def listener_callback(self, msg):
        #----------MESSAGE ARRAY----------
        print(f"2st Array: {list(msg.data)}")

        #----------MESSAGE TYPE----------
        message = UInt8MultiArray()
        message.data = [0, 0, 0, 0, 0]

        #----------MESSAGE ARRAY----------
        for i in range(5):
            if (msg.data[i] <= 5):
                message.data[i] = 0
            else:
                message.data[i] = msg.data[i] - 5
        print(f"Final Array: {list(message.data)}")
        

def main(args=None):
    rclpy.init(args=args)

    alpha_sub = Alpha_Subscriber()

    rclpy.spin(alpha_sub)

    alpha_sub.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: Pubsub_ws/Example_C_ws/src/sub.cpp
#include <memory>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/u_int8_multi_array.hpp"

using std::placeholders::_1;

class Alpha_Subscriber : public rclcpp::Node
{
  //----------DECLARATIONS----------
  rclcpp::Subscription<std_msgs::msg::UInt8MultiArray>::SharedPtr subscription_;

  public:
    //----------NODE DECLARATION----------
    Alpha_Subscriber() : Node("Subscriber_N")
    {
      subscription_ = this->create_subscription<std_msgs::msg::UInt8MultiArray>(
      "Topic_2", 10, std::bind(&Alpha_Subscriber::topic_callback, this, _1));
    }

  private:
    void topic_callback(const std_msgs::msg::UInt8MultiArray & msg) const
    {
      //----------SUBSCRIBER----------
      //----------MESSAGE ARRAY----------
      std::cout << "2nd Array: [";
      for (size_t i = 0; i < 5; ++i)
      {
        std::cout << static_cast<int>(msg.data[i]);
        if (i != msg.data.size() - 1)
        {
          std::cout << ", ";
        }
      }
      std::cout << "]" << std::endl;

      //----------FINAL MESSAGE----------
      //----------MESSAGE TYPE----------
      auto message = std_msgs::msg::UInt8MultiArray();
      message.data = {0, 0, 0, 0, 0};

      //----------MESSAGE ARRAY----------
      std::cout << "3rd Array: [";
      for (size_t i = 0; i < 5; ++i)
      {
        message.data[i] = msg.data[i] - 5;
        if (message.data[i] >= 30)
        {
          message.data[i] = 0;
        }

        std::cout << static_cast<int>(message.data[i]);
        if (i != message.data.size() - 1)
        {
          std::cout << ", ";
        }
      }
      std::cout << "]" << std::endl;
    }
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<Alpha_Subscriber>());
  rclcpp::shutdown();
  return 0;
}